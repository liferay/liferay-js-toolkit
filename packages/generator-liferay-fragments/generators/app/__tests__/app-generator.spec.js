const fs = require('fs');
const path = require('path');
const YeomanTest = require('yeoman-test');

function expectFile(base, _path) {
  const content = fs.readFileSync(path.join(base, _path), 'utf-8');
  return expect({
    _path: _path.split(path.sep).join('/'),
    content
  }).toMatchSnapshot();
}

function expectFiles(base, paths) {
  return paths.map(_path => expectFile(base, _path));
}

describe('fragment-generator', () => {
  it('generates a new project', () =>
    YeomanTest.run(path.join(__dirname, '..')).then(projectPath =>
      expectFiles(path.join(projectPath, 'sample-liferay-fragments'), [
        '.editorconfig',
        '.eslintrc',
        '.gitignore',
        '.yo-rc.json',
        'package.json',
        path.join('scripts', 'compress.js'),
        path.join('scripts', 'log.js')
      ])
    ));

  it('allows a custom repository name', () =>
    YeomanTest.run(path.join(__dirname, '..'))
      .withPrompts({ repositoryName: 'My Nice Custom Repository' })
      .then(projectPath =>
        expectFiles(path.join(projectPath, 'my-nice-custom-repository'), [
          'package.json'
        ])
      ));

  it('allows adding sample content', () =>
    YeomanTest.run(path.join(__dirname, '..'))
      .withPrompts({ addSampleContent: true })
      .then(projectPath => {
        expectFiles(path.join(projectPath, 'sample-liferay-fragments'), [
          'package.json'
        ]);

        expectFiles(
          path.join(
            projectPath,
            'sample-liferay-fragments',
            'src',
            'sample-collection',
            'sample-fragment'
          ),
          ['fragment.json', 'index.html', 'styles.css', 'main.js']
        );
      }));
});
