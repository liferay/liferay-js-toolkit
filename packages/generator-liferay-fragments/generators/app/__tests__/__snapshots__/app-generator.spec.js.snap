// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-generator allows a custom repository name 1`] = `
Object {
  "content": "{
  \\"name\\": \\"my-nice-custom-project\\",
  \\"description\\": \\"Liferay Fragments project\\",
  \\"version\\": \\"1.0.0\\",

  \\"engines\\": {
    \\"node\\": \\">= 8.0.0\\",
    \\"npm\\": \\">= 6.0.0\\"
  },

  \\"keywords\\": [
    \\"liferay\\",
    \\"liferay-fragments\\"
  ],

  \\"scripts\\": {
    \\"add-collection\\": \\"yo liferay-fragments:collection\\",
    \\"add-fragment\\": \\"yo liferay-fragments:fragment\\",
    \\"compress\\": \\"yo liferay-fragments:compress\\",
    \\"export\\": \\"yo liferay-fragments:export\\",
    \\"import\\": \\"yo liferay-fragments:import\\",
    \\"import:watch\\": \\"yo liferay-fragments:import --watch\\"
  }
}
",
  "filePath": "/package.json",
}
`;

exports[`app-generator allows a custom repository name 2`] = `
Object {
  "content": "# My Nice Custom Project

This project has been created with [generator-liferay-fragments][1]. You can read
how to manage this project in the [generator documentation][2].

[1]: https://www.npmjs.com/package/generator-liferay-fragments
[2]: https://www.npmjs.com/package/generator-liferay-fragments#usage
",
  "filePath": "/README.md",
}
`;

exports[`app-generator allows adding sample content 1`] = `
Object {
  "content": "{
  \\"name\\": \\"sample-liferay-fragments\\",
  \\"description\\": \\"Liferay Fragments project\\",
  \\"version\\": \\"1.0.0\\",

  \\"engines\\": {
    \\"node\\": \\">= 8.0.0\\",
    \\"npm\\": \\">= 6.0.0\\"
  },

  \\"keywords\\": [
    \\"liferay\\",
    \\"liferay-fragments\\"
  ],

  \\"scripts\\": {
    \\"add-collection\\": \\"yo liferay-fragments:collection\\",
    \\"add-fragment\\": \\"yo liferay-fragments:fragment\\",
    \\"compress\\": \\"yo liferay-fragments:compress\\",
    \\"export\\": \\"yo liferay-fragments:export\\",
    \\"import\\": \\"yo liferay-fragments:import\\",
    \\"import:watch\\": \\"yo liferay-fragments:import --watch\\"
  }
}
",
  "filePath": "/package.json",
}
`;

exports[`app-generator allows adding sample content 2`] = `
Object {
  "content": "# Sample Liferay Fragments

This project has been created with [generator-liferay-fragments][1]. You can read
how to manage this project in the [generator documentation][2].

[1]: https://www.npmjs.com/package/generator-liferay-fragments
[2]: https://www.npmjs.com/package/generator-liferay-fragments#usage
",
  "filePath": "/README.md",
}
`;

exports[`app-generator allows adding sample content 3`] = `
Object {
  "content": "{
  \\"cssPath\\": \\"styles.css\\",
  \\"htmlPath\\": \\"index.html\\",
  \\"jsPath\\": \\"main.js\\",
  \\"name\\": \\"Sample fragment\\",
  \\"type\\": \\"section\\"
}
",
  "filePath": "fragment.json",
}
`;

exports[`app-generator allows adding sample content 4`] = `
Object {
  "content": "<div class=\\"sample-fragment\\">
  Sample fragment
</div>
",
  "filePath": "index.html",
}
`;

exports[`app-generator allows adding sample content 5`] = `
Object {
  "content": ".sample-fragment {
  color: inherit;
}
",
  "filePath": "styles.css",
}
`;

exports[`app-generator allows adding sample content 6`] = `
Object {
  "content": "const content = fragmentElement.querySelector('.sample-fragment');
",
  "filePath": "main.js",
}
`;

exports[`app-generator generates a new project 1`] = `
Object {
  "content": "{
  \\"name\\": \\"sample-liferay-fragments\\",
  \\"description\\": \\"Liferay Fragments project\\",
  \\"version\\": \\"1.0.0\\",

  \\"engines\\": {
    \\"node\\": \\">= 8.0.0\\",
    \\"npm\\": \\">= 6.0.0\\"
  },

  \\"keywords\\": [
    \\"liferay\\",
    \\"liferay-fragments\\"
  ],

  \\"scripts\\": {
    \\"add-collection\\": \\"yo liferay-fragments:collection\\",
    \\"add-fragment\\": \\"yo liferay-fragments:fragment\\",
    \\"compress\\": \\"yo liferay-fragments:compress\\",
    \\"export\\": \\"yo liferay-fragments:export\\",
    \\"import\\": \\"yo liferay-fragments:import\\",
    \\"import:watch\\": \\"yo liferay-fragments:import --watch\\"
  }
}
",
  "filePath": "/package.json",
}
`;

exports[`app-generator generates a new project 2`] = `
Object {
  "content": "# Sample Liferay Fragments

This project has been created with [generator-liferay-fragments][1]. You can read
how to manage this project in the [generator documentation][2].

[1]: https://www.npmjs.com/package/generator-liferay-fragments
[2]: https://www.npmjs.com/package/generator-liferay-fragments#usage
",
  "filePath": "/README.md",
}
`;
